.github/workflows/deploy.yaml

```diff
@@ -1,43 +1,27 @@
 name: Deploy Docusaurus to GitHub Pages
 
 on:
-  # Trigger the workflow on pull requests and pushes to specific branches
-  pull_request:
   push:
     branches:
       - main
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
 # Concurrency configuration to manage parallel workflow runs
-#
-# Group composition: pages-<event_type>-<unique_identifier>
-# - event_type: 'pull_request', 'push', or 'workflow_dispatch'
-# - unique_identifier: PR number for PRs, branch ref for pushes/manual runs
-#
-# Examples of group names:
-# - PR #123: "pages-pull_request-123"
-# - Push to main: "pages-push-refs/heads/main"
-# - Manual run on main: "pages-workflow_dispatch-refs/heads/main"
-#
-# Behavior:
-# - PRs: New commits cancel previous runs (cancel-in-progress: true)
-# - Main branch: Runs complete without cancellation (cancel-in-progress: false)
-# - Manual dispatch: Runs complete without cancellation (cancel-in-progress: false)
 concurrency:
-  group: pages-${{ github.event_name }}-${{ github.event_name == 'pull_request' && github.event.pull_request.number || github.ref }}
-  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+  group: production-deploy
+  cancel-in-progress: true
 
 jobs:
   build:
     name: Build Docusaurus
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
         with:
           fetch-depth: 0
```

Comment:
Generally, I don't think you want `fetch-depth: 0` as it will download the entire git history when you probably only need the most recent commit which is faster.